crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -43,154 +42,464 @@ mod validate_op_tests;
 
 #[instrument(skip(
     workspace,
-    space,
+    incoming_dht_ops_sender,
+    current_validation_dependencies,
     trigger_app_validation,
-    sys_validation_trigger,
+    trigger_self,
     network
 ))]
-pub async fn sys_validation_workflow<Network: HolochainP2pDnaT + Clone + 'static>(
+pub async fn sys_validation_workflow<
+    Network: HolochainP2pDnaT + Clone + 'static,
+    Sender: DhtOpSender + Send + Sync + Clone + 'static,
+>(
     workspace: Arc<SysValidationWorkspace>,
-    space: Arc<Space>,
+    incoming_dht_ops_sender: Sender,
+    current_validation_dependencies: Arc<Mutex<ValidationDependencies>>,
     trigger_app_validation: TriggerSender,
-    sys_validation_trigger: TriggerSender,
+    trigger_self: TriggerSender,
     network: Network,
 ) -> WorkflowResult<WorkComplete> {
-    let complete =
-        sys_validation_workflow_inner(workspace, space, network, sys_validation_trigger).await?;
+    // Run the actual sys validation using data we have locally
+    let outcome_summary =
+        sys_validation_workflow_inner(workspace.clone(), current_validation_dependencies.clone())
+            .await?;
+
+    // trigger app validation to process any ops that have been processed so far
+    if outcome_summary.accepted > 0 {
+        tracing::debug!("Sys validation accepted {} ops", outcome_summary.accepted);
+
+        trigger_app_validation.trigger(&"sys_validation_workflow");
+    }
+
+    // Now go to the network to try to fetch missing dependencies
+    let network_cascade = Arc::new(workspace.network_and_cache_cascade(network));
+    let missing_action_hashes = current_validation_dependencies.lock().get_missing_hashes();
+    let num_fetched: usize = futures::stream::iter(missing_action_hashes.into_iter().map(|hash| {
+        let network_cascade = network_cascade.clone();
+        let current_validation_dependencies = current_validation_dependencies.clone();
+        async move {
+            match network_cascade.retrieve(hash, Default::default()).await {
+                Ok(Some((record, source))) => {
+                    let mut deps = current_validation_dependencies.lock();
+
+                    // If the source was local then that means some other fetch has put this action into the cache,
+                    // that's fine we'll just grab it here.
+                    if deps.insert(record, source) {
+                        1
+                    } else {
+                        0
+                    }
+                }
+                Ok(None) => {
+                    // This is fine, we didn't find it on the network, so we'll have to try again.
+                    // TODO This will hit the network again fairly quickly if sys validation is triggered again soon.
+                    //      It would be more efficient to wait a bit before trying again.
```

Comment:
This will need resolving. It's not too serious because it's hitting the fetch pool which should ignore duplicate requests and has its own back-off. Probably best not to rely on that detail here and actually back-off

Reply:
So this is retriggered based on the conductor's sys validation retry delay?